### DISCUSSION TITLE
Set secrets via the API?

### DISCUSSION THREAD

**Discussion by beckermr** [cirruslabs/cirrus-ci-docs#1201] [Q&A] [answered]
Is there a way to set secrets via the API?

I see some graphQL API endpoints, but it is not obvious to me how I might use them to properly encrypt the value.

**Comment by fkorotkov** [comment, marked as answer]
Hey @beckermr,

There is a way to create a new secret via API. You can check how [we do it in the frontend](https://github.com/cirruslabs/cirrus-ci-web/blob/6812226afdd510d87eb2bb9755947c422c0b5249/src/components/settings/OwnerSecuredVariables.tsx#L51-L53).

But I think you might be interested in [`updateSecuredOwnerVariable` mutation](https://github.com/cirruslabs/cirrus-ci-web/blob/6812226afdd510d87eb2bb9755947c422c0b5249/schema.gql#L418) which allows to update secrets by name. Note that you'll need [admin API rights](https://cirrus-ci.org/api/#authorization) to perform this mutation.

**Reply by beckermr** [reply]
Hmmmm ok. I honestly have no idea and you have far exceeded my knowledge of graphql and whatever .tsx is. I'll have to hack at it.

**Reply by beckermr** [reply]
I figured this out! Thank you!
